pymeasure/instruments/kuhneelectronic/kusg245_250a.py

```diff
@@ -0,0 +1,305 @@
+from pymeasure.instruments.validators import strict_discrete_set
+from pymeasure.instruments.validators import truncated_range, truncated_discrete_set
+
+
+class Kusg245_250A(Instrument):
+    """Represents KU SG 2.45 250 A the 2.45 GHz ISM-Band Microwave Generator
+    and provides a high-level interface for interacting with the instrument.
+
+    :param power_limit: power set-point limit in Watts.
+                        See :attr:`~.Kusg245_250A.power_setpoint`
+                        and :meth:`~.Kusg245_250A.tune()`.
+
+    Usage example:
+
+    .. code-block:: python
+
+        from pymeasure.instruments.kuhneelectronic import Kusg245_250A
+
+        generator = Kusg245_250A("ASRL3::INSTR", power_limit=100) # limits the output
+                                                                  # power set-point to 100 W
+
+        generator.external_enabled = False      # biasing and RF output controlled by serial comm
+        generator.power = 20                    # Sets the output power to 20 Watts
+        generator.bias_enabled = True           # Enables amplifier biasing
+        generator.rf_enabled = True             # Enables the RF output
+
+        p_fwd = generator.power_forward         # Reads forward power in Watts
+        p_rev = generator.power_reverse         # Reads reflected power in Watts
+    """
+
+    def __init__(self, adapter, name="KU SG 2.45 250 A", power_limit=250, **kwargs):
+        kwargs.setdefault("baud_rate", 115200)
+        kwargs.setdefault("read_termination", "\r")
+        kwargs.setdefault("write_termination", "\r")
+
+        super().__init__(adapter, name, **kwargs)
+
+        assert 0 < power_limit <= 250
+        self._power_limit = power_limit
+        self.power_setpoint_values = [0, power_limit]
+
+    version = Instrument.measurement("v", """Readout of the firmware version.""")
+
+    voltage_5v = Instrument.measurement(
+        "5",
+        """Readout of internal 5V supply voltage in Volts.""",
+        get_process=lambda v: 103.0 / 4700.0 * v,
+    )
+
+    voltage_32v = Instrument.measurement(
+        "8",
+        """Readout of 32V supply voltage in Volts.""",
+        get_process=lambda v: 1282.0 / 8200.0 * v,
+    )
+
+    power_forward = Instrument.measurement("6", """Readout of forward power in Watts.""")
+
+    power_reverse = Instrument.measurement("7", """Readout of reverse power in Watts.""")
+
+    temperature = Instrument.measurement(
+        "T", """Readout of temperature sensor near the final transistor in °C."""
+    )
+
+    external_enabled = Instrument.control(
+        "r?",
+        "%s",
+        """Control the whether the amplifier enabling is done
+        via external inputs on 8-pin connector
+        or via the serial interface (boolean).
+        """,
+        validator=strict_discrete_set,
+        values=[False, True],
+        set_process=lambda v: {True: "R", False: "r"}[v],
+        get_process=lambda v: bool(v)
```

Comment:
Why not do this just like `freq_steps_fine_enabled` (replacing the values in the dict accordingly)?

Reply:
The reason is in different protocol of setter and getter commands:
- To enable (disable) that function, user must send "R" ("r") commands.
- The status is checked by "r?" command but answer is "1" ("0"). 

Therefore, special `set_process` and `get_process` are used.

The same logic is in `bias_enabled`, `rf_enabled` and `pulse_mode_enabled` properties.

Reply:
oh, for... :cursing_face: 
I see, that makes sense! :-)